crates/bevy_tasks/src/single_threaded_task_pool.rs

```diff
@@ -198,15 +223,92 @@ impl TaskPool {
     }
 }
 
-/// An empty task used in single-threaded contexts.
-///
-/// This does nothing and is therefore safe, and recommended, to ignore.
-#[derive(Debug)]
-pub struct FakeTask;
+/// A handle to a task running on this thread. Can be awaited to obtain the output of the task.
+#[cfg(not(target_arch = "wasm32"))]
+pub type LocalTask<T> = crate::Task<T>;
 
-impl FakeTask {
-    /// No op on the single threaded task pool
+/// A handle to a task running on this thread. Can be awaited to obtain the output of the task.
+#[cfg(target_arch = "wasm32")]
+pub struct LocalTask<T>(Rc<InnerTask<T>>);
+
+#[cfg(target_arch = "wasm32")]
+type Panic = Box<dyn Any + Send>;
+
+#[cfg(target_arch = "wasm32")]
+struct InnerTask<T> {
+    value: Cell<Option<Result<T, Panic>>>,
+    waker: Cell<Option<Waker>>,
+}
```

Comment:
I was hoping I could get away with using `Cell`/`Rc` for this since it's nearly free at runtime, but on second thought this probably wouldn't be that useful since the handle is `!Send`, so you can't store it in the world even if the code is single-threaded. We'll likely have to use a one-shot channel for this